### DISCUSSION TITLE
QuartzSchedulerThread: Usefulness of GetRandomizedIdleWaitTime()

### DISCUSSION THREAD

**Discussion by drieseng** [quartznet/quartznet#1406] [General]
When we acquired zero triggers, we wait a certain time (unless the schedule changed, or the scheduler was paused or stopped) before we make another attempt to acquire triggers.

Instead of just use **IdleWaitTime** to determine how long we wait, we randomize the "wait time" by subtracting a random number of milliseconds (with a maximum of 20% of the total milliseconds of **IdleWaitTime**) from **IdleWaitTime**.

Is there really a benefit to this? I see a minor cost, and very little benefit.

@lahma Can you shed some light on this?

**Comment by lahma** [comment]
I can't remember the specifics but I don't see this as a big issue. Usually for retry-kind-of-logic we want some randomization, but not sure if what's the case here.
